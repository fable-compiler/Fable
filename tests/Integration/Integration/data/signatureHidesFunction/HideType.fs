module HideType

open Fable.Core

/// This type should be hidden by the signature file
type public Foobar =
    | Foo of int
    | Bar of string

let someFunction () =
    JS.console.log "meh"
