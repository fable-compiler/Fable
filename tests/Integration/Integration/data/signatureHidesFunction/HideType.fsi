module HideType

val someFunction: unit -> unit
