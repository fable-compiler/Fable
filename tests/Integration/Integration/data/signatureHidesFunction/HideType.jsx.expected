import { Union } from "./fable_modules/fable-library-js.4.13.0/Types.js";
import { union_type, string_type, int32_type } from "./fable_modules/fable-library-js.4.13.0/Reflection.js";
import { some } from "./fable_modules/fable-library-js.4.13.0/Option.js";

class Foobar extends Union {
    constructor(tag, fields) {
        super();
        this.tag = tag;
        this.fields = fields;
    }
    cases() {
        return ["Foo", "Bar"];
    }
}

function Foobar_$reflection() {
    return union_type("HideType.Foobar", [], Foobar, () => [[["Item", int32_type]], [["Item", string_type]]]);
}

export function someFunction() {
    console.log(some("meh"));
}

